legal-api/src/legal_api/services/business_details_version.py

```diff
@@ -34,9 +34,68 @@
 )
 
 
-class VersionedBusinessDetailsService:
+class VersionedBusinessDetailsService:  # pylint: disable=too-many-public-methods
     """Provides service for getting business details as of a filing."""
 
+    @staticmethod
+    def get_revision(filing_type, filing_id, business_id):
+        """Consolidates based on filing type upto the given transaction id of a filing."""
+        business = Business.find_by_internal_id(business_id)
+        filing = Filing.find_by_id(filing_id)
+
+        revision_json = {}
+        revision_json['filing'] = {}
+        if filing_type == 'incorporationApplication':
+            revision_json['filing'] = \
+                VersionedBusinessDetailsService.get_ia_revision(filing, business)
+        elif filing_type == 'changeOfDirectors':
+            pass
+        elif filing_type == 'changeOfAddress':
+            pass
+        elif filing_type == 'annualReport':
+            pass
+        elif filing_type == 'correction':
+            pass
+        elif filing_type == 'alteration':
```

Comment:
Even though you dont have reports for other type of filings like specialResolution, voluntary dissolution, change of name etc. I think we have to handle that here